### ISSUE TITLE
DataLocationProxy is broken

### ISSUE THREAD

**Issue by dready92** [quirkey/sammy#16]
Hello,

your Sammy.DataLocationProxy is broken : unfortunately jQuery (1.4.2) fires the setData event before actually updating the data store. So when you try a simple :

```
$.sammy('#main').bind("location-changed", function() {
    $.sammy('#main').log(  $.sammy('#main').getLocation()  );  
});
```

you get the old location...

**Comment by quirkey** [comment]
Looking into this - it must be a recent change in jQuery because this was definitely working with 1.4. thanks for the bug reports! I appreciate it.

**Comment by quirkey** [comment]
Actually fix the DataLocationProxy issue with accurate tests. Closed by 82c2927f973a59558ff30ee95ef1b3ecb31d9009
